pkg/v1/providers/tests/unit/avi_test.go

```diff
@@ -0,0 +1,261 @@
+		"$.kind":          "ClusterRole",
+		"$.metadata.name": "ako-cr",
+	}))
+	assertFoundOne(matchers.FindDocsMatchingYAMLPath(output, map[string]string{
+		"$.kind":          "ClusterRoleBinding",
+		"$.metadata.name": "ako-crb",
+	}))
+	assertFoundOne(matchers.FindDocsMatchingYAMLPath(output, map[string]string{
+		"$.kind":          "StatefulSet",
+		"$.metadata.name": "ako",
+	}))
+	assertFoundOne(matchers.FindDocsMatchingYAMLPath(output, map[string]string{
+		"$.kind":          "Service",
+		"$.metadata.name": "ako-operator-controller-manager-metrics-service",
+	}))
+	assertFoundOne(matchers.FindDocsMatchingYAMLPath(output, map[string]string{
+		"$.kind":          "ClusterRole",
+		"$.metadata.name": "ako-operator-manager-bootstrap",
+	}))
+	assertFoundOne(matchers.FindDocsMatchingYAMLPath(output, map[string]string{
+		"$.kind":          "ClusterRoleBinding",
+		"$.metadata.name": "ako-operator-bootstrap-rolebinding",
+	}))
+	assertFoundOne(matchers.FindDocsMatchingYAMLPath(output, map[string]string{
+		"$.kind":          "Deployment",
+		"$.metadata.name": "ako-operator-controller-manager",
+	}))
+	assertFoundOne(matchers.FindDocsMatchingYAMLPath(output, map[string]string{
+		"$.kind":          "Secret",
+		"$.metadata.name": AviAdminCredentialName,
+	}))
+	assertFoundOne(matchers.FindDocsMatchingYAMLPath(output, map[string]string{
+		"$.kind":          "Secret",
+		"$.metadata.name": AviCaName,
+	}))
+}
+
+var _ = Describe("AKO-operator Ytt Templating", func() {
+
+	var paths []string
+	var baseVal yttValues
+
+	BeforeEach(func() {
+		paths = []string{
+			filepath.Join(yamlRoot, "config_default.yaml"),
+			filepath.Join("./fixtures/tkr-bom-v1.21.1.yaml"),
+			filepath.Join("./fixtures/tkg-bom-v1.4.0.yaml"),
+			filepath.Join(yamlRoot, "ytt"),
+		}
+
+		baseVal = map[string]string{
+			// required fields
+			"TKG_DEFAULT_BOM":    "tkg-bom-v1.4.0.yaml",
+			"KUBERNETES_RELEASE": "v1.21.2---vmware.1-tkg.1",
+
+			// required fields to enable AVI
+			"PROVIDER_TYPE":                 "vsphere",
+			"TKG_CLUSTER_ROLE":              "management",
+			"AVI_ENABLE":                    "true",
+			"AVI_CONTROL_PLANE_HA_PROVIDER": "true",
+
+			// required vsphere configurations
+			"VSPHERE_USERNAME":           "user_blah",
+			"VSPHERE_PASSWORD":           "pass_1234",
+			"VSPHERE_SERVER":             "vmware-tanzu.com",
+			"VSPHERE_DATACENTER":         "vmware-tanzu-dc.com",
+			"VSPHERE_RESOURCE_POOL":      "myrp",
+			"VSPHERE_FOLDER":             "ds0",
+			"VSPHERE_SSH_AUTHORIZED_KEY": "ssh-rsa AAAA...+M7Q== vmware-tanzu.local",
+			"VSPHERE_INSECURE":           "true",
+
+			// required avi related values
+			"AVI_USERNAME":              AviUsername,
+			"AVI_PASSWORD":              AviPassword,
+			"AVI_CA_NAME":               AviCaName,
+			"AVI_CA_DATA_B64":           AviCaDataB64,
+			"AVI_ADMIN_CREDENTIAL_NAME": AviAdminCredentialName,
+		}
+	})
+
+	When("basic values are provided", func() {
```

Comment:
Basic value with avi enabled?

Reply:
yes, we have negative test for `AVI_ENABLED = false` down below